t/Nessy/daemon_dies_after_request.t

```diff
@@ -11,31 +14,70 @@ use AnyEvent;
 my $client = Nessy::Test::Client->new(url => 'http://localhost/');
 ok($client, 'created new client');
 
-my $w;
-$w = AnyEvent->io(fh => $client->daemon_to_client_sock,
-             poll => 'r',
-             # when the client sends the daemon a lock request, close
-             # the socket to simulate the daemon going away
-             cb => sub {
-                        ok(1, 'Got message from client to make a lock');
-                        $client->daemon_to_client_sock->close(),
-                        undef $w;
-                      },
+subtest 'client does not block when daemon dies' => sub {
+    plan tests => 4;
+
+    my $w;
+    $w = AnyEvent->io(fh => $client->daemon_to_client_sock,
+                 poll => 'r',
+                 # when the client sends the daemon a lock request, close
+                 # the socket to simulate the daemon going away
+                 cb => sub {
+                            ok(1, 'Got message from client to make a lock');
+                            $client->daemon_to_client_sock->close(),
+                            undef $w;
+                          },
+            );
+
+    local $SIG{ALRM} = sub { die "timed out" };
+    alarm(30);
+    my $warning_message;
+    local $SIG{__WARN__} = sub { $warning_message = shift };
+    my $claim_name = 'foo';
+    my $claim = eval { $client->claim($claim_name) };
+    alarm(0);
+
+    ok(! $claim, "couldn't make a claim when the daemon closed its socket");
+    ok(!$@, 'no exception');
+    like($warning_message,
+        qr(claim $claim_name at .* failed: Connection reset by peer),
+        'warning message');
+};
+
+subtest 'daemon restarts after going away' => sub {
+    plan tests => 5;
+
+    # when it gets the claim message, send it back as successful
+    my $respond_successfully = sub {
+                my($watcher, $message_hash) = @_;
+                my $message = Nessy::Client::Message->new(%$message_hash);
+                ok($message, 'Got message from client to make a lock');
```

Comment:
I think the test name should probably just be "received a message" because I think it's trigger for both `claim` and `release` below.